### PR TITLE
Fixed #6316 encoding will be applied to asXml

### LINKED ISSUE #6316: WSResponse.asXml() not work with no UTF-8 encoding
I use WSClient to access a remote service. I got the following response:

Transfer-Encoding: chunked
Content-Type: application/xml;charset=euc-kr

When I use
     Document doc = response.asXml() ;
for this resposne, I got:

application - org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 81; Invalid byte 1 of 1-byte UTF-8 sequence.
...
Caused by: org.xml.sax.SAXParseException: Invalid byte 1 of 1-byte UTF-8 sequence.
        at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
        at play.libs.XML.fromInputSource(XML.java:76)
        at play.libs.XML.fromInputStream(XML.java:56)
        at play.libs.ws.ahc.AhcWSResponse.asXml(AhcWSResponse.java:127)
 
It is right that at column 81 starting the Korean character.

If I use response.getBody() to get the string representation, and then pass the string to Jaxb unmarshall, everything works as exptected.

### PR DESCRIPTION
**Pull request by schmitch**
## Fixes

Fixes #6316

## Purpose

Adds the correct content encoding to WSResponse asXml()

### CODE REVIEW

File: framework/src/play-integration-test/src/test/scala/play/it/libs/WSSpec.scala [hunk 1]
```diff
@@ -44,6 +41,15 @@ trait WSSpec extends PlaySpecification with ServerIntegrationSpecification {
 
   val foldingSink = Sink.fold[ByteString, ByteString](ByteString.empty)((state, bs) => state ++ bs)
 
+  val isoString = {
```
Comment (schmitch) [inline framework/src/play-integration-test/src/test/scala/play/it/libs/WSSpec.scala]: not sure if that works? anybody got a better idea?

**Comment by mkurz** [comment]
@schmitch The tests you added [fail](https://travis-ci.org/playframework/playframework/jobs/144136507#L2718-L2719):
> [error]   x response asXml with correct contentType
> [error]    'Hello �' is not equal to 'Hello €' (WSSpec.scala:220)

**Comment by marcospereira** [comment]
LGTM.

**Comment by gmethvin** [comment]
Thanks @schmitch

**Comment by mkurz** [comment]
Should we backport this?

**Comment by schmitch** [comment]
actually I already working on the back port (running the tests).

**Pull request merged**